tezos/api/src/ffi.rs

```diff
@@ -562,38 +562,88 @@ pub type Json = String;
 pub struct JsonRpcRequest {
     pub body: Json,
     pub context_path: String,
+    pub meth: RpcMethod,
+    pub content_type: Option<String>,
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
-pub struct JsonRpcResponse {
+pub struct HelpersPreapplyResponse {
     pub body: Json
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum ProtocolRpcResponse {
+    RPCConflict(Option<String>),
+    RPCCreated(Option<String>),
+    RPCError(Option<String>),
+    RPCForbidden(Option<String>),
+    RPCGone(Option<String>),
+    RPCNotContent,
+    RPCNotFound(Option<String>),
+    RPCOk(String),
+    RPCUnauthorized,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum RpcMethod {
+    DELETE,
+    GET,
+    PATCH,
+    POST,
+    PUT,
+}
+
+impl TryFrom<&str> for RpcMethod {
+    type Error = &'static str;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        let upper = s.to_uppercase();
+        match upper.as_ref() {
+            "DELETE" => Ok(RpcMethod::DELETE),
+            "GET" => Ok(RpcMethod::GET),
+            "PATCH" => Ok(RpcMethod::PATCH),
+            "POST" => Ok(RpcMethod::POST),
+            "PUT" => Ok(RpcMethod::PUT),
+            _ => Err("Invalid RPC method"),
```

Comment:
@tizoc please, add `s`  to `Err("Invalid RPC method")` like
`Err(format!("Invalid RPC method: {}", s))`

Reply:
Done.